### ISSUE TITLE
The problem with RGB lamps

### ISSUE THREAD

**Issue by vring0** [adizanni/floor3d-card#23]
When the state of the RGB lamp changes from RGB color to WW / WC (warm and cool) the color temperature of the lamp. The changes are not visible on the plan, for this you need to refresh the browser page.

https://www.youtube.com/watch?v=X8KZ7-sdJO0

**Comment by adizanni** [comment]
Yes you  are right I do not manage the temperature. I would classify it as a feature request. Are you ok ?

**Comment by vring0** [comment]
yes, please as a feature request.

**Comment by adizanni** [comment]
Hello I've implemented a management of the temperature that I hope will be compatible with the majority of light devices. I've seen around that the it is somehow complex to take into account all temperature features. So far I'm managing the rgb mode and the color_temp mode as separate indipendent features. Please test release 0.9.17 and tell me if you are ok with the feature

**Comment by vring0** [comment]
> Hello I've implemented a management of the temperature that I hope will be compatible with the majority of light devices. I've seen around that the it is somehow complex to take into account all temperature features. So far I'm managing the rgb mode and the color_temp mode as separate indipendent features. Please test release 0.9.17 and tell me if you are ok with the feature

The light is not updated dynamically, only after reloading the Chrome browser page.

**Comment by adizanni** [comment]
Hello Vring0, please let me know if you have installed the latest version and tested with it. If you need to refresh the page in order to see the change in a light, it may also be due to HA not sending out the event of a state change to the card. I keep that I have some difficult to test all combination of light attributes change as I do not have all type of devices. Please post as much info as possible (console logs from the browser, errors in the core supervisor console, etc.)

**Comment by vring0** [comment]
![image](https://user-images.githubusercontent.com/34909289/126458495-759368c5-f70d-4ff2-8e1a-bc1261258060.png)
I am using the latest version. When I switch the lights, I get this error all the time. In version 0.9.15 there were no problems with this.

**Comment by adizanni** [comment]
Ok,  I can reproduce the bug. Something's still wrong with the temperature management

**Comment by adizanni** [comment]
Can you please upgrade to the latest release v0.9.18 and verify if the bug is corrected ?

**Comment by adizanni** [comment]
If you are not activating the shadow parameter (setting it to 'yes') the performances should not be better or worse than a few weeks ago. There is not a lot I can do to improve performance, the heavy lifting is done by a js module (three.js)  I have no control on (it is considered as one of the best open source project to render and animate 3D models. Of course if your model weighs 100Mb and you bind 100 entities, and you activate the shadows, etc. the performances will suffer....

**Comment by vring0** [comment]
I also have one last question about another problem. My walls are not transparent. But light passes through them. And some strange glare appeared on objects, even if the light was turned off.


Light passes through walls, example:
![image](https://user-images.githubusercontent.com/34909289/128127572-15b8367b-1cde-4d6b-870d-587edfbffa85.png)
Glare of light, example:
![image](https://user-images.githubusercontent.com/34909289/128127651-96fd1790-7362-4e2e-9a7a-2283799aad55.png)

**Comment by adizanni** [comment]
Hello Vring0, since version 0.9.12, I  have introduced a Directional Light to complement the ambient light. Without Directional Light the scene was too dark and object details would not appear, also textures were not rendered. I did not find a way to use ambient light to make the scened better defined. I can put a parameter to switch off the directional light so  that you can see the difference.

**Comment by adizanni** [comment]
Sorry I did not answer all your questions. Since versiobn 0.9.18 I have also introduced the shadow parameter.  This prevents the light beam to flood through walls. This makes the scene more realistinc, but it has quite a significant impact on the processing especially with your complex objects this might become quite heavy. Please test it (put shadow to 'yes") and let me know what you think.

**Comment by vring0** [comment]
![image](https://user-images.githubusercontent.com/34909289/128291281-412a0cb8-c3b5-4f12-959b-68e564849c6c.png)
ceiling lights stopped shining down

**Comment by adizanni** [comment]
what is the object that is bound to the ceiling light ? If you have put the half sphere, it should project some light down, if not, light may be shaded by parts of the light objects itself. Shadow parameter is still a work in progress

**Comment by vring0** [comment]
For the test, I use all the objects of the luminaire. I will wait for a stable release. I can attach my model for testing.

**Comment by adizanni** [comment]
Try to bind only the sphere object to the light (not the group of objects) and tell me what is the results. If you attach your model it can help me for the performance test.

**Comment by vring0** [comment]
I did that too. There is no light down.
my files: [livingroom2.zip](https://github.com/adizanni/floor3d-card/files/6938966/livingroom2.zip)

**Comment by adizanni** [comment]
I tried with the following code:

```yaml
type: custom:floor3d-card
entities:
  - entity: light.gateway_light_7811dcb06bfe
    type3d: light
    object_id: Sphere_104_328
    light:
      lumens: '400'
object_groups:
  - object_group: ''
name: vring0
path: /local/test-vring0/livingroom2/
objfile: livingroom.obj
mtlfile: livingroom.mtl
shadow: 'yes'
```
With only one entity the model is loading fast enough 1.5 sec

this is the result with the light with shadow = 'yes' and using the sphere as object:

![testvring0](https://user-images.githubusercontent.com/35622920/128383507-4ed5bcc0-2568-498f-a1b2-06a745ec42f8.png)

**Comment by vring0** [comment]
![image](https://user-images.githubusercontent.com/34909289/128399494-e49772f6-e448-444e-ae61-19b892ebeaa0.png)
I have a problem with these lamps.

**Comment by adizanni** [comment]
Unfortunately my calculation for the positioning of the light is not good for some lights objects as the light source falls inside the object and when you activate the shadow, the object is shadowing the light. If you  build those spotlights by yourself, you may try to do a cilinder open downtowards or with a transparent bottom. Also I suggest you give a look to my new plugin for Sweethome3D that makes the export easier (object_id will be more stable as they are based on object names). You can find it here: https://github.com/adizanni/ExportToHASS

**Comment by adizanni** [comment]
Look at these models: https://www.polantis.com/ikea/hektar-suspension

**Comment by vring0** [comment]
![image](https://user-images.githubusercontent.com/34909289/128413077-2e9960da-e850-4dec-9aa7-10a55f77becb.png)
Please write how to import into Sweet Home 3D?

**Comment by adizanni** [comment]
You download the .obj format unzip it (with the .mtl file0 and you  right click on the futniture section of sweethome3d and do import furniture  -> choose model -> select the obj file and import it to the right Light section. Be careful that dimensio and orientation may be wrong, you need to adjust

**Comment by adizanni** [comment]
I close this issue as it is longstanding and I all feeback are positive about the lights. In case some problem persist please open another issue
